internal/handler/handler.go

```diff
@@ -201,56 +241,36 @@ func (h *Handler) Process(body []byte, headers map[string]string) (response help
 			return helpers.Response{StatusCode: http.StatusUnprocessableEntity}, nil
 		}
 
-		pCtx.Owner = e.Repo.Owner.Login
-		pCtx.Repository = e.Repo.Name
 		pCtx.BaseRef = helpers.NormaliseRefPtr(*e.PullRequest.Base.Ref)
 		pCtx.HeadRef = helpers.NormaliseRefPtr(*e.PullRequest.Head.Ref)
 		pCtx.HeadSHA = e.PullRequest.Head.SHA
 
-		if pCtx.Promoter == nil {
```

Comment:
These are important to be kept as they are used to support passing a static promoter and only if that is not set that dynamic promotion is enabled. Otherwise we need to handover the "is dynamic promotion enabled" all the way to handler and then use that to gate the dynamic promotion spawning.

This also enables the tests runtime as dynamic promotion is not always wanted/desired/supported.